lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb

```diff
@@ -28,13 +83,79 @@ def initialize(response_builder, dispatcher)
       def on_call_node_enter(node)
         content = extract_test_case_name(node)
 
-        return unless content
+        if content
+          append_document_symbol(
+            name: content,
+            selection_range: range_from_node(node),
+            range: range_from_node(node),
+          )
+        end
+
+        extract_callbacks(node)
+      end
+
+      private
+
+      sig { params(node: Prism::CallNode).void }
+      def extract_callbacks(node)
+        receiver = node.receiver
+        return if receiver && !receiver.is_a?(Prism::SelfNode)
+
+        message_value = node.message
+        callbacks = [MODEL_CALLBACKS, CONTROLLER_CALLBACKS, JOB_CALLBACKS]
+
+        return unless callbacks.any? { |callback| callback.include?(message_value) }
+
+        block = node.block
 
+        if block
+          append_document_symbol(
+            name: "#{message_value}(<anonymous>)",
+            range: range_from_location(node.location),
+            selection_range: range_from_location(block.location),
+          )
+          return
+        end
+
+        arguments = node.arguments&.arguments
+        return unless arguments&.any?
+
+        arguments.each do |argument|
+          if argument.is_a?(Prism::SymbolNode)
+            name = argument.value
+            next unless name
+
+            append_document_symbol(
+              name: "#{message_value}(#{name})",
+              range: range_from_location(argument.location),
+              selection_range: range_from_location(T.must(argument.value_loc)),
+            )
+          elsif argument.is_a?(Prism::StringNode)
+            name = argument.content
+            next if name.empty?
+
+            append_document_symbol(
+              name: "#{message_value}(#{name})",
+              range: range_from_location(argument.location),
+              selection_range: range_from_location(argument.content_loc),
+            )
+          end
+        end
+      end
+
+      sig do
+        params(
+          name: String,
+          range: RubyLsp::Interface::Range,
+          selection_range: RubyLsp::Interface::Range,
+        ).void
+      end
+      def append_document_symbol(name:, range:, selection_range:)
         @response_builder.last.children << RubyLsp::Interface::DocumentSymbol.new(
-          name: content,
-          kind: LanguageServer::Protocol::Constant::SymbolKind::METHOD,
-          selection_range: range_from_node(node),
-          range: range_from_node(node),
+          name: name,
+          kind: RubyLsp::Constant::SymbolKind::METHOD,
```

Comment:
This affects the icon of the symbol.

I chose the same approach as the base Ruby LSP. The options are listed in [this link](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolKind) — I don't think it's optimal to use any other config.